collect/collect.go

```diff
@@ -256,16 +296,26 @@ func (c *InventoryCollection) Upload() error {
 	}
 	defer res.Body.Close()
 
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		log.Error().Err(err).Int("status", res.StatusCode).Msg("reading response")
+		return errors.Wrap(err, "reading response")
+	}
+
 	if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusCreated {
-		body, err := io.ReadAll(res.Body)
-		if err != nil {
-			log.Error().Err(err).Int("status", res.StatusCode).Msg("reading response")
-			return errors.Wrap(err, "reading response")
-		}
 		log.Error().Int("status", res.StatusCode).Str("body", string(body)).Msg("")
 		return errors.New("upload failed")
 	}
 
+	metaDataResponse := &uploadResponse{}
+	if err := json.Unmarshal(body, metaDataResponse); err != nil {
```

Comment:
It may be a little bit more efficient to parse the stream something like
```suggestion
	if err := json.NewDecoder(body).Decode(metaDataResponse); err != nil {
```

Then you wont have to read it all into memory before starting to parse it - not a big deal with this size but anyway.